src/main/kotlin/com/zepben/evolve/cim/iec61970/base/core/IdentifiedObject.kt

```diff
@@ -62,30 +61,75 @@ abstract class IdentifiedObject(mRID: String = "") {
      */
     fun getName(type: String, name: String): Name? = _names?.getByTypeAndName(type, name)
 
-    fun addName(name: Name): IdentifiedObject {
-        require(name.identifiedObject === this) { "Attempting to add a Name to ${typeNameAndMRID()} that does not reference this identified object" }
+    /**
+     * The individual name information of the identified object.
+     *
+     * @param name the name of the required [Name]
+     * @param type the required [NameType]
+     * @return The [Name] with the specified [name] if it exists, otherwise null
+     */
+    fun getName(type: NameType, name: String): Name? = _names?.getByTypeAndName(type.name, name)
+
+    /**
+     * All name information of the identified object given a [NameType].
+     *
+     * @param type the required [NameType]
+     * @return List of [Name] with the specified [type] if it exists, otherwise null
+     */
+    fun getNames(type: NameType): List<Name>? = if (_names?.any { it.type == type } == true) _names?.filter { it.type == type } else null
```

Comment:
todo: These functions run the risk of double processing the entire collection of names if there is only one matching name and it is the last one in the collection. This can be done in a single pass using `takeUnless` or `takeIf`. e.g.

```
_names?.filter { it.type == type }.takeUnless { it.isEmpty() }
```
or
```
_names?.filter { it.type == type }.takeIf { it.isNotEmpty() }
```

Reply:
Done